integration-libs/opf/base/core/services/opf-cart-handler.service.ts

```diff
@@ -165,7 +172,16 @@ export class OpfCartHandlerService {
       userId: this.cartHandlerState.userId,
       extraData: { active: true },
     });
-    return this.checkStableCart().pipe(
+    return forkJoin({
+      previousCartStable: this.checkStableCart(
+        this.cartHandlerState.previousCartId
+      ),
+      cartStable: this.checkStableCart(this.cartHandlerState.cartId),
+    }).pipe(
+      switchMap(() => this.activeCartFacade.getActiveCartId()),
+      filter((cartId) => cartId === this.cartHandlerState.previousCartId),
+      take(1),
```

Comment:
we first make sure the 2 carts that were just loaded and unloaded (previousCartId becomes the activeCart, cartId becomes a stale cart) are stable.
Important as stale cart gets deleted afterward, we want to avoid any 'cart not found'.
take(1) needed as getActiveCartId is a stream